openlibrary/plugins/openlibrary/js/carousels.js

```diff
@@ -53,13 +53,41 @@ var Carousel = {
             });
         }
 
+        function showSlide(el, iterate) {
+            setTimeout(function() {
+                $('.slick-active').each(function (index) {
+                    var $currentEl = $(this);
+                    (function ($current) {
+                        setTimeout(function () {
+                            $current.addClass('opacity-show');
```

Comment:
I'm not a big fan of js animations as usually there is an equivalent css animation.

What's stopping you from making a CSS only solution (applying transition to the img tag and relying on the existing Slick css class and its removal?

Reply:
Thanks for the review. Unfortuantely we have to work with what slick gives us here. The problem with trying to use the slick classes to accomplish this is that the ‘.slick-active’ class will be added before the lazy loading of images are complete. Another problem with that is that at certain breakpoints slick-active is added to a couple of images that are still hidden in the next slide and not yet lazy loaded. 

I am actually just adding and removing classes via jquery and the transitions are pure css. I am utilizing the slick events (init, lazyLoaded) to add and remove classes only after we know the images are loaded.

Reply:
(https://github.com/kenwheeler/slick/blob/a2aa3fec335c50aceb58f6ef6d22df8e5f3238e1/slick/slick.js#L1548 is where I found about those classes)

Reply:
Ah yeah I see. Yeah I think you are right adding opacity 0 to slick loading will probably do the trick.